### PR TITLE
[SMALLFIX] Adding mount table journal compaction.

### PR DESCRIPTION
**Pull request by jsimsa**
This method fixes an issue with journal compaction, that would cause mount points disappear across multiple restarts.

### CODE REVIEW

**Comment by gpang** [comment]
LGTM

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 1]
```diff
@@ -2019,43 +2020,45 @@ private void mountFromEntry(AddMountPointEntry entry)
...
+      String defaultUfsPath = MasterContext.getConf().get(Constants.UNDERFS_ADDRESS);
+      UnderFileSystem defaultUfs = UnderFileSystem.get(defaultUfsPath, MasterContext.getConf());
+      if (defaultUfs.exists(PathUtils.concatPath(defaultUfsPath, alluxioPath.getPath()))) {
+        throw new IOException(
+            ExceptionMessage.MOUNT_PATH_SHADOWS_DEFAULT_UFS.getMessage(alluxioPath));
+      }
+
       // Configure the ufs properties, and update the mount options with the configured properties.
       ufs.configureProperties();
       options.setProperties(ufs.getProperties());
```
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: Do we need the try catch still?

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 2]
```diff
@@ -344,6 +344,7 @@ public void processJournalEntry(JournalEntry entry) throws IOException {
 
   @Override
   public void streamToJournalCheckpoint(JournalOutputStream outputStream) throws IOException {
+    mMountTable.streamToJournalCheckpoint(outputStream);
```
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: Is it okay to write all the mount info before the inode tree info?

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 1 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: no, the addition has been moved after this operation so there is no need to delete the mount point if the ufs operation throws an exception

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 2 above)
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: yes

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**